### PR TITLE
[1.3.2] add ecm shell clear

### PR DESCRIPTION
**Pull request by huangKai-2323**
### What is the purpose of the change

add ecm shell clear

### Related issues/PRs

Related issues: #4104
Related pr:#4103

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/scala/org/apache/linkis/ecm/server/util/ThreadUtils.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.slf4j.Logger;
+
+public class ThreadUtils extends ApplicationContextEvent {
+
+    private static final Logger logger = LoggerFactory.getLogger(ThreadUtils.class);
+    public static final String shellPath = Configuration.getLinkisHome() + "/admin/";
+
+    public static ExecutionContextExecutorService executors =
+            Utils.newCachedExecutionContext(5, "alert-pool-thread-", false);
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/scala/org/apache/linkis/ecm/server/util/ThreadUtils.java:39]: rename `alert-pool-thread- `  to ` clean-ec-pool-thred`?

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/scheduled/ScheduledTask.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component
+public class ScheduledTask {
```
[the remark this answers is not in the document]
Comment (huangKai-2323) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/scheduled/ScheduledTask.java:35]: done

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/scala/org/apache/linkis/ecm/server/util/ThreadUtils.java (hunk 1 above)
Comment (huangKai-2323) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/scala/org/apache/linkis/ecm/server/util/ThreadUtils.java:39]: done

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/scheduled/EcmClearTask.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  private Logger logger = LoggerFactory.getLogger(EcmClearTask.class);
+
+  private class CleanExpiredThread implements Runnable {
+    @Override
+    public void run() {
+      logger.info("Start to linkis-ec-clear shell");
+      List<String> cmdlist = new ArrayList<>();
+      cmdlist.add("sh");
+      cmdlist.add(ThreadUtils.shellPath + "linkis-ec-clear.sh");
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/scheduled/EcmClearTask.java:45]: missing  linkis-ec-clear.sh file

Comment (huangKai-2323) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/scheduled/EcmClearTask.java:45]: done

File: linkis-dist/package/admin/linkis-ec-clear.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#The cleaning date must be passed in. And ecm will print the actual effective date log.
+end_date=`date -d '-90 days'`
+end_date_t=`date -d '-90 days' +%s`
+echo "Clean up data before this date:$end_date"
+#Directory read ecm engine cache variables
+prefix=/data/bdp/linkis/
```
Comment (peacewong) [inline linkis-dist/package/admin/linkis-ec-clear.sh:22]: need to update install.sh replace prefix

Comment (huangKai-2323) [reply linkis-dist/package/admin/linkis-ec-clear.sh]: done

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**